firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -912,7 +920,8 @@ Map<String, Object> serialize(T object, ErrorPath path) {
     }
 
     private void applyFieldAnnotations(Field field) {
```

Comment:
I don't think that's the right place to check for getters, instead we should modify applyGetterAnnotations, I would really like for someone on the firestore team to take a look tbh.